### PR TITLE
Make TonY client log layout more organized

### PR DESCRIPTION
**Pull request by burgerkingeater**
This PR contains following changes
- Sort updated tasks based on status, name, and index. 
- Display all tasks' status when application is progressing so that users can check which task is running. 
- When the application is progressing, task status is being printed out in a simplified way where no URL is included for cleanness.
- Display all tasks' status along with their URL when application finishes.

The new log layout looks like following:
```
25-03-2021 19:49:07 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:07 INFO  TonyClient:961 - ------  Application (re)starts, status of ALL tasks ------
25-03-2021 19:49:07 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:07 INFO  TonyClient:1036 - RUNNING, chief, 0, some url
25-03-2021 19:49:07 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:07 INFO  TonyClient:1036 - RUNNING, ps, 0, some url
25-03-2021 19:49:07 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:07 INFO  TonyClient:1036 - RUNNING, worker, 0, some url
25-03-2021 19:49:07 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:07 INFO  TonyClient:1036 - RUNNING, worker, 1, some url
25-03-2021 19:49:34 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:34 INFO  TonyClient:968 - ------ Task Status Updated ------
25-03-2021 19:49:34 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:34 INFO  TonyClient:1086 - FAILED: chief [0] ps [0] 
25-03-2021 19:49:34 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:34 INFO  TonyClient:1086 - RUNNING: worker [0, 1] 
25-03-2021 19:49:35 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:35 INFO  TonyClient:968 - ------ Task Status Updated ------
25-03-2021 19:49:35 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:35 INFO  TonyClient:1086 - FAILED: chief [0] ps [0] 
25-03-2021 19:49:35 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:35 INFO  TonyClient:1086 - FINISHED: worker [0, 1] 
25-03-2021 19:49:36 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:36 INFO  TonyClient:984 - -----  Application finished, status of ALL tasks -----
25-03-2021 19:49:36 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:36 INFO  TonyClient:1036 - FAILED, chief, 0, some url
25-03-2021 19:49:36 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:36 INFO  TonyClient:1036 - FAILED, ps, 0, some url
25-03-2021 19:49:36 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:36 INFO  TonyClient:1036 - FINISHED, worker, 0, some url
25-03-2021 19:49:36 PDT mnist-avro-distributed INFO - 2021-03-26 02:49:36 INFO  TonyClient:1036 - FINISHED, worker, 1, some url
```

### CODE REVIEW

**Comment by oliverhu** [comment]
are we printing out status of all workers for every update?

**Comment by burgerkingeater** [comment]
@oliverhu yes, to give user a holistic view. But print only when status changes.

**Comment by burgerkingeater** [comment]
@oliverhu updated to just print out updated tasks in the middle and print out all tasks at the end.

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java [hunk 1]
```diff
@@ -891,32 +898,71 @@ public boolean monitorApplication() throws YarnException, IOException, Interrupt
...
+   * [TaskInfo] name: worker, index: 3, url: url status: FAILED
+   * [TaskInfo] name: worker, index: 1, url: url status: SUCCEEDED
+   * [TaskInfo] name: worker, index: 5, url: url status: FINISHED
+   * [TaskInfo] name: ps, index: 0, url: url status: RUNNING
+   * [TaskInfo] name: worker, index: 0, url: url status: RUNNING
+   * [TaskInfo] name: worker, index: 2, url: url status: RUNNING
+   * [TaskInfo] name: worker, index: 4, url: url status: RUNNING
+   */
+  @VisibleForTesting
+  static List<TaskInfo> sortTaskInfo(Collection<TaskInfo> tasks) {
```
Comment (oliverhu) [inline tony-core/src/main/java/com/linkedin/tony/TonyClient.java:916]: why not make TaskInfo Comparable?

**Comment by burgerkingeater** [comment]
Closing this for now and will resume on this later.

**Comment by burgerkingeater** [comment]
@oliverhu updated the PR, please take a look.

**Review by oliverhu — APPROVED** [review]

**Pull request merged**